design-system/Avatar.svelte

```diff
@@ -56,6 +56,15 @@
       }
     }
   }
+
+  function validImgUrl(image_url: string): boolean {
+    const http = new XMLHttpRequest();
+
+    http.open("HEAD", image_url, false);
+    http.send();
+
+    return http.status !== 404;
+  }
```

Comment:
This approach to detecting a valid image has two issues. First, it only checks for 404 status codes and not for other status codes or invalid content types like the server sending JSON. Second, this is not async and blocks the render thread. This means that when it takes longer to get a response the app becomes unresponsive.

I have a PR (https://github.com/radicle-dev/radicle-upstream/pull/2602) that chooses a different approach.

Reply:
Makes sense. Merged ✔